### ISSUE TITLE
Mlflow x Autogluon

### ISSUE THREAD

**Issue by noamanemobidata** [awslabs/autogluon#1404]
Hi! there is any plan to support Mlflow  for logging and loading Autogluon models ?

**Comment by eavidan** [comment]
I was looking for that as well. I assume this is a contribution needed on mlflow side.
In the mean while was able to create a custom `PythonModel` for my Autogluon `TabularPredictor`. 
Maybe you can use it as well. 

```python
class AutogluonModel(mlflow.pyfunc.PythonModel):

    def load_context(self, context):
        self.predictor = TabularPredictor.load(context.artifacts.get("predictor_path"))

    def predict(self, context, model_input):
        return self.predictor.predict(model_input)
```

Then I can use it to log my model

```python
model = AutogluonModel()
predictor_path = "/path/to/my/predictor"
artifacts = {"predictor_path": predictor_path}
conda_env = {
        'channels': ['conda-forge'],
        'dependencies': [
            f'python={python_version()}',
            'pip'],
        'pip': [
            f'mlflow=={mlflow.__version__}',
            f'autogluon=={autogluon_version.__version__}',
            f'cloudpickle=={cloudpickle.__version__}'
        ],
        'name': 'mlflow-env'
    }
mlflow.pyfunc.log_model(artifact_path="model", python_model=model,
                            artifacts=artifacts,
                            conda_env=conda_env)
```

**Comment by MBlancoC** [comment]
@eavidan Very useful, thanks a lot!

**Comment by michaeljoyce217** [comment]
**TL/DR: I did much the same thing but I'm hung up on a technicality with an AutoGluon timeseries functionality.**

I built a similar PythonModel for my Autogluon.timeseries model using individual models like DeepAR, rather than the elaborate TimeSeriesPredictor. I know you can restrict its functionality, but I want to average 3 models to control outliers and sometimes gain from negative and positive errors cancelling out.

However, I really want to add "known_covariates" and my colleague and I are struggling! The issue is that the ".fit" has to be in "train.py" in our MLflow system, while the ".predict" has to be in "inference.py" with a "deploy.py" step in the middle that registers the model from a Databricks experiment to Azure based MLflow.

To be honest, this has me at the edge of my abilities, but .fit takes a list parameter "known_covariates_names" and .predict uses that under the hood to extract columns from the dataframe parameter "known_covariates." We have tried a lot of different methods to have the wrapper absorb this list, pass it and implement it correctly.

Any suggestions welcome! If we succeed, I'll post back here.
